app/controllers/searches/users_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class Searches::UsersController < ApplicationController
+  before_action :authenticate_user!
+
+  # ユーザーの検索一覧を表示
+  def index
+    params[:q].each do |key, value|
+      params[:q].delete(key) if value.blank?
+    end
+
+    if params[:q].present?
```

Comment:
多分今のコードなら、nil、または、hashが空ではないという判定の方が適切かと思います。
理由は下に記載

```suggestion
    if params[:q].nil? || !params[:q].empty?
```